apps/src/codegen.js

```diff
@@ -266,6 +270,10 @@ export function marshalNativeToInterpreter(interpreter, nativeVar, nativeParentO
       nativeFunc: nativeVar,
       nativeParentObj: nativeParentObj,
     };
+    if (exports.asyncFunctionList.indexOf(nativeVar) !== -1) {
+      // Mark if this should be nativeIsAsync:
+      makeNativeOpts.nativeIsAsync = true;
+    }
```

Comment:
I believe @pcardune is trying to gradually move this sort of behavior into `CustomMarshalingInterpreter` and `CustomMarshaler`, and gradually out of `codegen`. I don't see a super-clean way to do that yet here - maybe that's a good clean-up for a later PR.